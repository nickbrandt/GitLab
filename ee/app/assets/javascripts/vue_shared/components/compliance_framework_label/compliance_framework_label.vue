<script>
import { GlLabel } from '@gitlab/ui';
import { validateHexColor } from '~/lib/utils/color_utils';

export default {
  components: {
    GlLabel,
  },
  props: {
    color: {
      type: String,
      required: true,
      validator: (color) => validateHexColor(color),
    },
    description: {
      type: String,
      required: true,
    },
    name: {
      type: String,
      required: true,
    },
  },
};
</script>

<template>
  <gl-label size="sm" :background-color="color" :title="name" :description="description" />
</template>
