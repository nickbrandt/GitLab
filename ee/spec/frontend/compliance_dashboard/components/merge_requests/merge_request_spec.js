import { GlAvatarLink, GlAvatar } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';

import MergeRequest from 'ee/compliance_dashboard/components/merge_requests/merge_request.vue';
import ComplianceFrameworkLabel from 'ee/vue_shared/components/compliance_framework_label/compliance_framework_label.vue';
import { complianceFramework } from 'ee_jest/vue_shared/components/compliance_framework_label/mock_data';
import { createMergeRequest } from '../../mock_data';

describe('MergeRequest component', () => {
  let wrapper;

  const findAuthorAvatarLink = () => wrapper.find('.issuable-authored').find(GlAvatarLink);
  const findComplianceFrameworkLabel = () => wrapper.findComponent(ComplianceFrameworkLabel);

  const createComponent = (mergeRequest) => {
    return shallowMount(MergeRequest, {
      propsData: {
        mergeRequest,
      },
      stubs: {
        CiIcon: {
          props: { status: Object },
          template: `<div class="ci-icon">{{ status.group }}</div>`,
        },
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('when there is a merge request', () => {
    const mergeRequest = createMergeRequest({
      props: {
        compliance_management_framework: complianceFramework,
      },
    });

    beforeEach(() => {
      wrapper = createComponent(mergeRequest);
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the title', () => {
      expect(wrapper.text()).toContain(mergeRequest.title);
    });

    it('renders the issuable reference', () => {
      expect(wrapper.text()).toContain(mergeRequest.issuable_reference);
    });

    it('renders the author avatar', () => {
      expect(findAuthorAvatarLink().find(GlAvatar).exists()).toEqual(true);
    });

    it('renders the author name', () => {
      expect(findAuthorAvatarLink().text()).toEqual(mergeRequest.author.name);
    });

    it('renders the compliance framework label', () => {
      const { color, description, name } = complianceFramework;

      expect(findComplianceFrameworkLabel().props()).toStrictEqual({
        color,
        description,
        name,
      });
    });
  });

  describe('when there is a merge request without a compliance framework', () => {
    const mergeRequest = createMergeRequest();

    beforeEach(() => {
      wrapper = createComponent(mergeRequest);
    });

    it('does not render the compliance framework label', () => {
      expect(findComplianceFrameworkLabel().exists()).toBe(false);
    });
  });
});
