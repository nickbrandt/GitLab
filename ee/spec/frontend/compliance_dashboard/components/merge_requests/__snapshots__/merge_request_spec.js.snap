// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MergeRequest component when there is a merge request matches the snapshot 1`] = `
<div
  class="gl-grid-col-start-1 gl-border-b-solid gl-border-b-1 gl-border-b-gray-100 gl-p-5"
  data-testid="merge-request"
>
  <div>
    <a
      class="gl-text-gray-900 gl-font-weight-bold"
      href="/h5bp/html5-boilerplate/-/merge_requests/1"
    >
      
      Merge request 1
    
    </a>
  </div>
   
  <span
    class="gl-text-gray-500"
  >
    project!1
  </span>
   
  <span
    class="issuable-authored gl-text-gray-500 gl-display-inline-flex gl-align-items-center"
  >
    
    - created by:
    
    <gl-avatar-link-stub
      class="gl-display-inline-flex gl-align-items-center gl-ml-3 gl-text-gray-900 author-link js-user-link"
      data-name="User 1"
      data-user-id="1"
      href="http://localhost:3000/user-1"
      title="User 1"
    >
      <gl-avatar-stub
        alt="avatar"
        class="gl-mr-2"
        entityid="1"
        entityname="User 1"
        shape="circle"
        size="16"
        src="https://1"
      />
       
      <span>
        User 1
      </span>
    </gl-avatar-link-stub>
  </span>
   
  <div>
    <compliance-framework-label-stub
      color="#009966"
      description="General Data Protection Regulation"
      name="GDPR"
    />
  </div>
</div>
`;
